test/test.js

```diff
@@ -683,6 +683,20 @@ suite('SystemJS Standard Tests', function() {
     });
   });
 
+  test('Source maps', function () {
+    return System.import('tests/x-error.js')
+    .catch(function (error) {
+      ok(error.stack.includes('x-error.js'));
+    });
+  });
```

Comment:
The tests are a little bit awkward because the address is calculated. But because the code is using `runInThisContext` (https://nodejs.org/api/vm.html#vm_script_runinthiscontext_options) and the source address is passed in as the filename. We can throw an error in the dependency and check the stacktrace to verify the address is correct.

This first test verifies the change didn't break old functionality.